File: src/test/java/com/thebluehats/server/game/enchants/PeroxideTest.java
package com.thebluehats.server.game.enchants;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

import com.thebluehats.server.game.enchants.args.common.PotionEffectArgs;
import com.thebluehats.server.game.managers.combat.templates.ArrowHitPlayerTemplate;
import com.thebluehats.server.game.managers.combat.templates.EventTemplate;
import com.thebluehats.server.game.managers.combat.templates.PlayerHitPlayerTemplate;

import org.bukkit.entity.Player;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;
import org.junit.Test;

public class PeroxideTest {
    @Test
    public void DamageeGetsRegenerationWhenHit() {
        Peroxide enchant = new Peroxide(
                new EventTemplate[] { new ArrowHitPlayerTemplate(), new PlayerHitPlayerTemplate() });
        Player player = mock(Player.class);

        int duration = 2;
        int amplifier = 1;

        enchant.execute(new PotionEffectArgs(player, duration, amplifier));

        verify(player).addPotionEffect(new PotionEffect(PotionEffectType.REGENERATION, duration * 20, amplifier));
    }
}

File: minecraft-server/src/test/kotlin/com/thebluehats/server/game/enchants/PeroxideTest.kt
package com.thebluehats.server.game.enchants

import com.thebluehats.server.game.managers.enchants.processedevents.DamageEventEnchantData
import io.kotest.core.spec.style.DescribeSpec
import io.mockk.every
import io.mockk.mockk
import io.mockk.verify
import org.bukkit.entity.Player
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType

class PeroxideTest : DescribeSpec({
    describe("peroxide") {
        it("should give regen when hit") {
            val peroxide = Peroxide(mockk(), mockk())
            val effect = PotionEffect(PotionEffectType.REGENERATION, 8 * 20, 1)

            val damageeMock = mockk<Player> {
                every { addPotionEffect(effect, true) } returns true
            }

            val data = mockk<DamageEventEnchantData> {
                every { level } returns 3
                every { damagee } returns damageeMock
            }

            peroxide.execute(data)

            verify { damageeMock.addPotionEffect(effect, true) }
        }
    }
})